examples/Training/java/src/training/ROIs.java

```diff
@@ -126,97 +117,115 @@ private void createROIs()
     {
         DataManagerFacility dm = gateway.getFacility(DataManagerFacility.class);
         ROIFacility roifac = gateway.getFacility(ROIFacility.class);
-        Roi roi = new RoiI();
-        roi.setImage(image.asImage());
-        Rectangle rect = new RectangleI();
-        rect.setX(omero.rtypes.rdouble(10));
-        rect.setY(omero.rtypes.rdouble(10));
-        rect.setWidth(omero.rtypes.rdouble(10));
-        rect.setHeight(omero.rtypes.rdouble(10));
-        rect.setTheZ(omero.rtypes.rint(0));
-        rect.setTheT(omero.rtypes.rint(0));
-        roi.addShape(rect);
-        //Create a rectangular shape
-        rect = new RectangleI();
-        rect.setX(omero.rtypes.rdouble(10));
-        rect.setY(omero.rtypes.rdouble(10));
-        rect.setWidth(omero.rtypes.rdouble(10));
-        rect.setHeight(omero.rtypes.rdouble(10));
-        rect.setTheZ(omero.rtypes.rint(1));
-        rect.setTheT(omero.rtypes.rint(0));
-        roi.addShape(rect); //Add the shape
+
+        ROIData data = new ROIData();
+        data.setImage(image.asImage());
+        
+        //Create Rectangle ShapeData
+        RectangleData rectangleData = new RectangleData(10, 10, 10, 10);
+        rectangleData.setZ(0);
+        rectangleData.setT(0);
+        data.addShapeData(rectangleData);
+        
+        rectangleData = new RectangleData(10, 10, 10, 10);
+        rectangleData.setZ(0);
```

Comment:
have removed the Z and T information for the ellipse ROI, and have also added a documentation line to highlight the use case! Please check.